### ISSUE TITLE
Ansel globally intercepts all keyboard shortcuts, even the unassigned ones

### ISSUE THREAD

**Issue by koiuo** [aurelienpierreeng/ansel#203]
## Description of the bug

**To Reproduce**

1. Focus Ansel window
2. Press any key combination not normally handled by Ansel (I noticed this with Brightness up/down, Volume up/down)
3. Observe Ansel reacting to these shortcuts

## Expected behavior

Ansel ignores unassigned key-bindings

## Context

**Screenshots**

![image](https://github.com/aurelienpierreeng/ansel/assets/787439/e46d4253-6e0f-4d41-a8d6-1647521baf59)

## System

* darktable version : 26acfc25137d264cd39efdccc3de99a24e746f3f (built from source)
* OS                : Linux 6.4.4
* Linux - Distro    : NixOS
* Desktop           : KDE + i3

**Comment by koiuo** [comment]
@aurelienpierre ah, sorry, I failed to mention. For me key strokes captured by Ansel are no longer captured by KDE.
For instance, if I press Brightness Up/Down while Ansel is focused, it does not control the screen brightness. I have to switch focus to some other window for that key to work.

> tells you if you made a mistake

Perhaps, that's a useful info for a developer, but this is not how other software conventionally behaves. I'd expect Ansel to silently ignore key strokes, that are not mapped to any function.

**Comment by aurelienpierre** [comment]
So that's another bug of the great MIDI turducken…

**Comment by pedrorrodriguez** [comment]
> @aurelienpierre ah, sorry, I failed to mention. For me key strokes captured by Ansel are no longer captured by KDE. For instance, if I press Brightness Up/Down while Ansel is focused, it does not control the screen brightness. I have to switch focus to some other window for that key to work.
> 
> > tells you if you made a mistake
> 
> Perhaps, that's a useful info for a developer, but this is not how other software conventionally behaves. I'd expect Ansel to silently ignore key strokes, that are not mapped to any function.

Maybe that should be a different issue?

**Comment by lukadh** [comment]
@koiuo, Please change the title of your issue to reflect 

> For me key strokes captured by Ansel are no longer captured by KDE.
For instance, if I press Brightness Up/Down while Ansel is focused, it does not control the screen brightness. I have to switch focus to some other window for that key to work.

**Comment by lukadh** [comment]
Also: Have you had the same issue with previoums builds of Ansel? Or could the behaviour be related to an OS update? Just to make sure the problem comes down to the "MIDI turducken"

**Comment by koiuo** [comment]
> Please change the title of your issue to reflect
@lukadh, not sure what's your expectation of the title, I gave it my best shot.

**Comment by aurelienpierre** [comment]
Brightness and volume Up/Down work here with Ansel master and AppImage on KDE. The soft does not show the feedback message and the OS is properly capturing them and acting on sound and brightness as expected. 

Perhaps we have something more contextual going on here.

**Comment by lukadh** [comment]
> @lukadh, not sure what's your expectation of the title, I gave it my best shot.

Yep, that's fine. I just wanted to make clear that your issue lies with the blocking of system shortcuts not the notification in Ansel.

> This was the only Ansel version I have tried.

Sorry I didn't realise earlier which version you are on! Please, try a more recent version. There have been several and different reports of issues with keyboard shortcuts for the version you are using. This lead to a series of commits fixing shortcuts. So updating might just do the trick for you.

**Comment by koiuo** [comment]
@lukadh, @aurelienpierre , I have built the latest master, and it did not change the behavior for me.

However, I have two new important finds:

1. the issue can't be reproduced with an external keyboard. Only keys *on the laptop* are intercepted
2. DarkTable 4.x also has this issue

I tried installing DarkTable 3.0.2 (just an arbitray old version I could locate for my OS), and it does NOT have this issue. Hence, this is a regression.
I installed DarkTable 3.0.2 with all its dependencies, hence this regression could be anywhere, but my hunch that it's in DarkTable because Ansel and DarkTable are the only apps with such sympotms for me.

I'll continue digging. I guess it would be helpful to locate a specific version or commit that introduced the regression, so I intend to focus on that. Let me know if anything else could be helpful.

**Comment by aurelienpierre** [comment]
The commit should be one related to e3da6ed5420321d83d48153488f8587d864f963f, replacing standard Gtk key shortcuts ("accelerators") by an in-house monstruosity meant to support gamepads and MIDI devices, at the cost of making the code unlegible and breaking basic expectations from typical desktop users using typical desktop devices.

At that point, I think the whole thing will have to be reverted to basic Gtk accels because the code is pure technical debt and really not sustainable long-term. See for yourself there : https://github.com/aurelienpierreeng/ansel/blob/master/src/gui/accelerators.c

**Comment by lukadh** [comment]
Since the shortcuts have been rewritten, @koiuo  can you confirm that your issue has been resolved?

**Comment by aurelienpierre** [comment]
Yeah, it's resolved now.
